tests/cloner_test.go

```diff
@@ -1019,19 +1020,37 @@ var _ = Describe("all clone tests", func() {
 						}
 					}
 
-					for _, dv := range dvs {
+					for _, dv := range targetDvs {
 						By("Verifying MD5 sum matches")
 						matchFile := filepath.Join(testBaseDir, "disk.img")
 						Expect(f.VerifyTargetPVCContentMD5(f.Namespace, utils.PersistentVolumeClaimFromDataVolume(dv), matchFile, md5sum[:32])).To(BeTrue())
+
+						pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dv.Namespace).Get(context.TODO(), dv.Name, metav1.GetOptions{})
+						Expect(err).ToNot(HaveOccurred())
+
+						if cloneSourcePod := pvc.Annotations[ctrl.AnnCloneSourcePod]; cloneSourcePod != "" {
+							By(fmt.Sprintf("Getting pod %s/%s", dv.Namespace, cloneSourcePod))
+							pod, err := f.K8sClient.CoreV1().Pods(dv.Namespace).Get(context.TODO(), cloneSourcePod, metav1.GetOptions{})
+							Expect(err).ToNot(HaveOccurred())
+							podsNodeName[pod.Spec.NodeName] = true
+						}
+
 						By("Deleting verifier pod")
 						err = f.K8sClient.CoreV1().Pods(f.Namespace.Name).Delete(context.TODO(), utils.VerifierPodName, metav1.DeleteOptions{})
 						Expect(err).ToNot(HaveOccurred())
 						_, err = utils.WaitPodDeleted(f.K8sClient, utils.VerifierPodName, f.Namespace.Name, verifyPodDeletedTimeout)
 						Expect(err).ToNot(HaveOccurred())
 					}
+
+					// All pods should be in the same node except when the map is empty in smart clone
```

Comment:
You should be able to use `cloneType` in this context to tell you if this is going to be smart clone or not
For example `cloneType != "snapshot"`

Reply:
Ok, I changed the condition to `if cloneType == "network"`